endpoints/openrtb2/amp_auction_test.go

```diff
@@ -395,103 +395,106 @@ func TestCCPAConsent(t *testing.T) {
 	}
 }
 
-func TestNoConsent(t *testing.T) {
-	// Build Request
-	bid, err := getTestBidRequest(true, nil, true, nil)
-	if err != nil {
-		t.Fatalf("Failed to marshal the complete openrtb.BidRequest object %v", err)
+func TestConsentWarnings(t *testing.T) {
+	type inputTest struct {
+		regs              *openrtb_ext.ExtRegs
+		invalidConsentURL bool
+		expectedWarnings  map[openrtb_ext.BidderName][]openrtb_ext.ExtBidderMessage
 	}
+	invalidConsent := "invalid"
 
-	// Simulated Stored Request Backend
-	stored := map[string]json.RawMessage{"1": json.RawMessage(bid)}
+	bidderWarning := openrtb_ext.ExtBidderMessage{
+		Code:    10003,
+		Message: "debug turned off for bidder",
+	}
+	invalidCCPAWarning := openrtb_ext.ExtBidderMessage{
+		Code:    10001,
+		Message: "Consent '" + invalidConsent + "' is not recognized as either CCPA or GDPR TCF.",
+	}
+	invalidConsentWarning := openrtb_ext.ExtBidderMessage{
+		Code:    10001,
+		Message: "CCPA consent is invalid and will be ignored. (request.regs.ext.us_privacy must contain 4 characters)",
+	}
+	tc2Wrnings := map[openrtb_ext.BidderName][]openrtb_ext.ExtBidderMessage{openrtb_ext.BidderReservedGeneral: {invalidCCPAWarning}}
+	tc3Wrnings := map[openrtb_ext.BidderName][]openrtb_ext.ExtBidderMessage{
+		openrtb_ext.BidderReservedGeneral:  {invalidCCPAWarning, invalidConsentWarning},
+		openrtb_ext.BidderName("appnexus"): {bidderWarning},
+	}
+	tc4Wrnings := map[openrtb_ext.BidderName][]openrtb_ext.ExtBidderMessage{openrtb_ext.BidderName("appnexus"): {bidderWarning}}
 
-	// Build Exchange Endpoint
-	mockExchange := &mockAmpExchange{}
-	endpoint, _ := NewAmpEndpoint(
-		mockExchange,
-		newParamsValidator(t),
-		&mockAmpStoredReqFetcher{stored},
-		empty_fetcher.EmptyFetcher{},
-		&config.Configuration{MaxRequestSize: maxSize},
-		newTestMetrics(),
-		analyticsConf.NewPBSAnalytics(&config.Analytics{}),
-		map[string]string{},
-		[]byte{},
-		openrtb_ext.BuildBidderMap(),
-	)
+	testData := []inputTest{
+		{nil, false, nil},
+		{nil, true, tc2Wrnings},
+		{&openrtb_ext.ExtRegs{USPrivacy: "invalid"}, true, tc3Wrnings},
+		{&openrtb_ext.ExtRegs{USPrivacy: "1NYN"}, false, tc4Wrnings},
```

Comment:
Nice. I think we can go one step further to using named field setters and embedding the expected warnings:
```
testData := []inputTest{
	{
		regs:              nil,
		invalidConsentURL: false,
		expectedWarnings:  nil,
	},
	{
		regs:              nil,
		invalidConsentURL: true,
		expectedWarnings: map[openrtb_ext.BidderName][]openrtb_ext.ExtBidderMessage{
			openrtb_ext.BidderReservedGeneral: {invalidCCPAWarning},
		},
	},
	{
		regs:              &openrtb_ext.ExtRegs{USPrivacy: "invalid"},
		invalidConsentURL: true,
		expectedWarnings: map[openrtb_ext.BidderName][]openrtb_ext.ExtBidderMessage{
			openrtb_ext.BidderReservedGeneral: {invalidCCPAWarning, invalidConsentWarning},
			openrtb_ext.BidderAppnexus:        {bidderWarning},
		},
	},
	{
		regs:              &openrtb_ext.ExtRegs{USPrivacy: "1NYN"},
		invalidConsentURL: false,
		expectedWarnings: map[openrtb_ext.BidderName][]openrtb_ext.ExtBidderMessage{
			openrtb_ext.BidderAppnexus: {bidderWarning},
		},
	},
}
```